runtime/vm/ContinuationHelpers.cpp

```diff
@@ -0,0 +1,167 @@
+		result = FALSE;
+		goto end;
+	}
+
+	memset(continuation, 0, sizeof(J9VMContinuation));
+
+#ifdef J9VM_INTERP_GROWABLE_STACKS
+#define VMTHR_INITIAL_STACK_SIZE ((vm->initialStackSize > (UDATA) vm->stackSize) ? vm->stackSize : vm->initialStackSize)
+#else
+#define VMTHR_INITIAL_STACK_SIZE vm->stackSize
+#endif
+
+	if ((stack = allocateJavaStack(vm, VMTHR_INITIAL_STACK_SIZE, NULL)) == NULL) {
+		vm->internalVMFunctions->setNativeOutOfMemoryError(currentThread, 0, 0);
+		result = FALSE;
+		goto end;
+	}
+
+#undef VMTHR_INITIAL_STACK_SIZE
+
+	continuation->stackObject = stack;
+	continuation->stackOverflowMark = continuation->stackOverflowMark2 = J9JAVASTACK_STACKOVERFLOWMARK(stack);
+
+	frame = ((J9SFJNINativeMethodFrame*)stack->end) - 1;
+	frame->method = NULL;
+	frame->specialFrameFlags = 0;
+	frame->savedCP = NULL;
+	frame->savedPC = (U_8*)(UDATA)J9SF_FRAME_TYPE_END_OF_STACK;
+	frame->savedA0 = (UDATA*)(UDATA)J9SF_A0_INVISIBLE_TAG;
+	continuation->sp = (UDATA*)frame;
+	continuation->literals = (J9Method*)0;
+	continuation->pc = (U_8*)J9SF_FRAME_TYPE_JNI_NATIVE_METHOD;
+	continuation->arg0EA = (UDATA*)&frame->savedA0;
+	continuation->stackObject->isVirtual = TRUE;
+
+	J9VMJDKINTERNALVMCONTINUATION_SET_VMREF(currentThread, continuationObject, continuation);
+
+	/* GC Hook to register Continuation object */
+end:
+	return result;
+}
+
+BOOLEAN
+enterContinuation(J9VMThread *currentThread, j9object_t continuationObject)
+{
+	BOOLEAN result = TRUE;
+	jboolean started = J9VMJDKINTERNALVMCONTINUATION_STARTED(currentThread, continuationObject);
+	J9VMContinuation *continuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(currentThread, continuationObject);
+
+	VM_ContinuationHelpers::swapFieldsWithContinuation(currentThread, continuation);
+	currentThread->currentContinuation = continuation;
+
+	if (started) {
+		/* resuming Continuation from yield */
+		VM_OutOfLineINL_Helpers::restoreInternalNativeStackFrame(currentThread);
+		VM_OutOfLineINL_Helpers::returnSingle(currentThread, JNI_TRUE, 1);
+
+		currentThread->returnValue = J9_BCLOOP_EXECUTE_BYTECODE;
+
+		c_cInterpreter(currentThread);
+
+		restoreCallInFrame(currentThread);
+	} else {
+		/* start new */
+		UDATA args[] = { (UDATA) continuationObject };
+		J9NameAndSignature executeNameAndSig = { (J9UTF8*)&execute_name, (J9UTF8*)&execute_sig };
+	
+		J9VMJDKINTERNALVMCONTINUATION_SET_STARTED(currentThread, continuationObject, JNI_TRUE);
+
+		runStaticMethod(currentThread, J9UTF8_DATA(&continuationClass_name), &executeNameAndSig, 1, args);
+	}
+
+	J9VMJDKINTERNALVMCONTINUATION_SET_FINISHED(currentThread, continuationObject, JNI_TRUE);
+
+	/* Swap to parent Continuation */
+	j9object_t parentContinuation = J9VMJDKINTERNALVMCONTINUATION_PARENT(currentThread, continuationObject);
+	if (NULL != parentContinuation) {
+		currentThread->currentContinuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(currentThread, parentContinuation);
+	} else {
+		currentThread->currentContinuation = NULL;
```

Comment:
instead of checking parent, just do ` 		currentThread->currentContinuation = NULL;` after `VM_ContinuationHelpers::swapFieldsWithContinuation(currentThread, continuation);` below